apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/profile/ProfileTaskExecutionService.java

```diff
@@ -108,36 +139,49 @@ public void run() {
     /**
      * stop profile task, remove context data
      */
-    private synchronized void stopCurrentProfileTask(ProfileTaskExecutionContext needToStop) {
+    synchronized void stopCurrentProfileTask(ProfileTaskExecutionContext needToStop) {
         // stop same context only
         if (needToStop == null || !taskExecutionContext.compareAndSet(needToStop, null)) {
             return;
         }
 
+        // current execution stop running
+        needToStop.setRunning(false);
+
         // remove task
         profileTaskList.remove(needToStop.getTask());
 
-        // TODO notify OAP current profile task execute finish
+        // notify profiling task has finished
+        ServiceManager.INSTANCE.findService(ProfileTaskChannelService.class).notifyProfileTaskFinish(needToStop.getTask());
     }
 
     @Override
     public void prepare() throws Throwable {
-
     }
 
     @Override
     public void boot() throws Throwable {
-
+        // init PROFILE_THREAD and start
+        profileThread = new ProfileThread();
+        profileThread.setDaemon(true);
+        profileThread.setName("PROFILE-MONITOR-THREAD");
```

Comment:
`PROFILE-MONITOR-THREAD` -> `PROFILING-THREAD` . We don't use the word `monitor`, because SkyWalking is monitoring many places :)